File: examples/java/advanced/error-handling/unrecoverable-error.java
try {
    // Your code here
} catch (StepFailedException e) {
    // Access error details from unrecoverable errors
    System.out.println("Execution terminated: " + e.getMessage());
} catch (InvokeFailedException e) {
    System.out.println("Invocation failed: " + e.getMessage());
}

File: examples/python/sdk-reference/error-handling/unrecoverable-error.py
from aws_durable_execution_sdk_python.exceptions import ExecutionError, InvocationError

try:
    # Your code here
    pass
except (ExecutionError, InvocationError) as e:
    # Access termination reason from unrecoverable errors
    print(f"Execution terminated: {e.termination_reason}")
